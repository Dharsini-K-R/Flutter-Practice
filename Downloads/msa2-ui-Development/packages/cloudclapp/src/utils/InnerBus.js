import EventEmitter from "events";

// List of all events available for usage with inner bus
const EVENTS = {
  LOGOUT: "LOGOUT",
  LOGIN: "LOGIN",
};

class InnerBus extends EventEmitter {
  evt = EVENTS;
}

export default new InnerBus();
