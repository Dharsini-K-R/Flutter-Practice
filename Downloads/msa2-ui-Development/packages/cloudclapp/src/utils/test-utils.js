import React from "react";
import { render } from "@testing-library/react";
import { renderHook, act } from "@testing-library/react-hooks";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { StylesProvider } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/core";
import SnackbarProvider from "cloudclapp/src/SnackbarProvider";
import createTheme from "cloudclapp/src/styles/theme";
import thunk from "redux-thunk";
const middlewares = [thunk];

const msaTheme = createTheme();

// Make generated classNames consistent for snapshots
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`;

const AllTheProviders = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={msaTheme}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </StylesProvider>
  );
};

const customRender = (component, options, initialRoute = "/") => {
  const componentWithRouter = (
    <Router basename={initialRoute}>{component}</Router>
  );

  return render(componentWithRouter, { wrapper: AllTheProviders, ...options });
};

const renderWithProvider = ({
  childComponent,
  mockStore,
  mockState = {},
  initialRoute,
}) => {
  const store = mockStore || configureMockStore(middlewares)(mockState);

  return {
    ...customRender(
      <Provider store={store}>{childComponent}</Provider>,
      null,
      initialRoute,
    ),
    store,
  };
};

const customRenderHook = (callback, options = {}) => {
  const wrapper = options.wrapper || AllTheProviders;

  return renderHook(callback, { ...options, wrapper });
};

// re-export everything
export * from "@testing-library/react";

// override render and renderHook methods
export {
  customRender as render,
  renderWithProvider,
  customRenderHook as renderHook,
  act as actOnHook,
  userEvent,
};
