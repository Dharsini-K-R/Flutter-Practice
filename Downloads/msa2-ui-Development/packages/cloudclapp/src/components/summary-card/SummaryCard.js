import React from "react";
import {
  Grid,
  Typography,
  Card,
  CardContent,
  Button,
  makeStyles,
  Tooltip,
} from "@material-ui/core";
import { useTranslation } from "react-i18next";
import SecurityStatusIcon from "cloudclapp/src/components/security-status-icon";
import { formatDateOrString } from "msa2-ui/src/utils/date";
import { CircularProgress } from "@material-ui/core";
import { ENVIRONMENT_SUMMARY_CARDS } from "cloudclapp/src/Constants";

const useStyles = makeStyles(({ palette, typography }) => {
  return {
    cardStyle: {
      height: "220px",
      padding: "5px",
      borderRadius: "10px",
      boxShadow:
        "0px 4px 24px rgba(49, 64, 90, 0.1), 0px 2px 8px rgba(178, 188, 206, 0.2)",
    },
    bodyWrapper: {
      alignItems: "center",
      height: "7em",
    },
    headerWrapper: {
      height: "2em",
    },
    flexSpaceBtw: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
    },
    opaqueIcon: {
      fontSize: "35px",
      opacity: "0.2",
    },
    bodyText: {
      color: typography.body1.color,
      fontWeight: "500",
    },
    securityStatus: (props) => ({
      fontSize: 40,
      color: props.statusColor,
    }),
    textSupport: {
      color: palette.text.support,
      marginTop: "10px",
      fontWeight: "600",
    },
    button: {
      color: palette.primary.main,
      "&.MuiButton-outlined": {
        border: `1px solid ${palette.primary.main}`,
      },
    },
    lastAdded: {
      color: palette.text.support,
      fontWeight: "600",
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-end",
    },
    iconWrapper: {
      width: 75,
    },
    fullWidth: {
      width: "100%",
    },
    loader: {
      height: 150,
    },
  };
});
const SummaryCard = ({
  title,
  icon: SummaryIcon,
  count = 0,
  prefix = "",
  noDataText,
  buttonText,
  buttonDisabled = false,
  disabledButtonText = "",
  onClickCallback,
  securityLabel,
  status,
  statusColor,
  updatedAt,
  isLoading = false,
  idPrefix = "",
}) => {
  const classes = useStyles({ statusColor });
  const { t } = useTranslation();
  return (
    <Card className={classes.cardStyle}>
      <CardContent>
        <Grid
          container
          className={`${classes.flexSpaceBtw} ${classes.headerWrapper}`}
        >
          <Typography variant="h5" id={`${idPrefix}_CARD_TITLE`}>
            {title}
          </Typography>
          <SummaryIcon
            className={classes.opaqueIcon}
            id={`${idPrefix}_CARD_ICON`}
          />
        </Grid>

        {isLoading ? (
          <Grid
            container
            alignItems="center"
            justifyContent="center"
            className={classes.loader}
          >
            <CircularProgress aria-label={t("Loading")} />
          </Grid>
        ) : (
          <>
            {title === ENVIRONMENT_SUMMARY_CARDS.SECURITY ||
            title === ENVIRONMENT_SUMMARY_CARDS.WEBAPPSECURITY ? (
              <Grid container className={classes.bodyWrapper}>
                <Grid
                  item
                  className={classes.iconWrapper}
                  container
                  alignContent="center"
                >
                  <SecurityStatusIcon
                    id={`${idPrefix}_CARD_SECURITY_STATUS_ICON`}
                    status={status}
                  />
                </Grid>
                <Grid item>
                  <Typography
                    id={`${idPrefix}_CARD_SECURITY_STATUS`}
                    variant="h5"
                    className={classes.securityStatus}
                  >
                    {securityLabel}
                  </Typography>
                </Grid>
              </Grid>
            ) : (
              <Grid container className={classes.bodyWrapper}>
                {count > 0 ? (
                  <Typography
                    variant="h2"
                    className={classes.bodyText}
                    id={`${idPrefix}_CARD_COUNT`}
                  >
                    {prefix}
                    {count}
                  </Typography>
                ) : (
                  <Typography
                    variant="body1"
                    className={classes.textSupport}
                    id={`${idPrefix}_CARD_NO_DATA`}
                  >
                    {noDataText}
                  </Typography>
                )}
              </Grid>
            )}

            <Grid container className={classes.flexSpaceBtw}>
              <Tooltip title={buttonDisabled ? disabledButtonText : undefined}>
                <Grid item xs>
                  {buttonText && (
                    <Button
                      variant="outlined"
                      id={`${idPrefix}_CARD_ACTION_BUTTON`}
                      className={classes.button}
                      onClick={onClickCallback}
                      disabled={buttonDisabled}
                    >
                      {buttonText}
                    </Button>
                  )}
                </Grid>
              </Tooltip>
              {updatedAt &&
              (count > 0 ||
                title === ENVIRONMENT_SUMMARY_CARDS.SECURITY ||
                title === ENVIRONMENT_SUMMARY_CARDS.WEBAPPSECURITY) ? (
                <Grid item xs className={classes.fullWidth}>
                  <Typography variant="body1" className={classes.lastAdded}>
                    {t("Last Added")}
                  </Typography>
                  <Typography
                    variant="body1"
                    className={classes.lastAdded}
                    id={`${idPrefix}_CARD_LAST_UPDATED_AT`}
                  >
                    {formatDateOrString(updatedAt, "dd MMM yyyy HH:mm:ss")}
                  </Typography>
                </Grid>
              ) : null}
            </Grid>
          </>
        )}
      </CardContent>
    </Card>
  );
};

export default SummaryCard;
