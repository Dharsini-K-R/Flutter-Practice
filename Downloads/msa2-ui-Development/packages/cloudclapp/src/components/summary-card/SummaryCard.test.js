import React from "react";
import { render } from "cloudclapp/src/utils/test-utils";
import SummaryCard from "./SummaryCard";

const TEST_ICON = "Test Icon";
const TestIcon = () => <div>{TEST_ICON}</div>;

const props = {
  title: "Deployments",
  icon: TestIcon,
  count: 2,
  buttonText: "Add Deployment",
  onClickCallback: jest.fn(),
  idPrefix: "ENVIRONMENT_DETAILS_DEPLOYMENTS",
};
describe("SummaryCard component", () => {
  it("renders correctly", () => {
    const { container, getByText } = render(<SummaryCard {...props} />);
    expect(getByText("Deployments")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("intially renders the loading spinner", () => {
    const { getByLabelText } = render(
      <SummaryCard {...props} isLoading={true} />,
    );
    const loader = getByLabelText("Loading");
    expect(loader).toBeDefined();
  });
});
