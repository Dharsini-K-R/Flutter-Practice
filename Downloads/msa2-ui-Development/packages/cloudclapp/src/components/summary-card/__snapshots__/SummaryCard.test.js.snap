// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCard component renders correctly 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-cardStyle MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiCardContent-root"
  >
    <div
      class="MuiGrid-root makeStyles-flexSpaceBtw makeStyles-headerWrapper MuiGrid-container"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
        id="ENVIRONMENT_DETAILS_DEPLOYMENTS_CARD_TITLE"
      >
        Deployments
      </h5>
      <div>
        Test Icon
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-bodyWrapper MuiGrid-container"
    >
      <h2
        class="MuiTypography-root makeStyles-bodyText MuiTypography-h2"
        id="ENVIRONMENT_DETAILS_DEPLOYMENTS_CARD_COUNT"
      >
        
        2
      </h2>
    </div>
    <div
      class="MuiGrid-root makeStyles-flexSpaceBtw MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button"
          id="ENVIRONMENT_DETAILS_DEPLOYMENTS_CARD_ACTION_BUTTON"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add Deployment
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
